[StaticPageGenerator] Fix generating static content for homepage

dvesh3 opened the pull request:
## Changes in this pull request  
Resolves #13353

## Additional info  
Update nginx config to have URI mapping and restart the docker container to test the fix.

dvesh3 committed:
[StaticPageGenerator] Fix generating static content for homepage

jdreesen commented:
The nginx config needs to be updated in the skeleton and demo, too.

tmannherz commented on bundles/CoreBundle/EventListener/Frontend/StaticPageGeneratorListener.php:
```diff
@@ -143,6 +143,11 @@ public function onKernelResponse(ResponseEvent $event)
             return;
         }
 
+        if (($contentType = $event->getResponse()->headers->get('Content-Type'))
```
FWIW, I think both the `onKernelResponse` _and_ `onKernelRequest` methods in this class should further restrict the requests that are processed by this listener. The restrictions I've added to an overridden class that extends this are:
```php
if ($request->getMethod() !== 'GET' || !in_array('text/html', $request->getAcceptableContentTypes())) {
    return;
}
```
You wouldn't ever want to return statically-generated page contents for non-GET requests or if the client doesn't accept html.

kingjia90 commented:
Tested and it works now, but in the case there's a page called `/home`, and both `root`(id:1) and `/home` do generate static pages, it wouldn't work for `/home` as it will be replaced by the content of `/`.

Probably it would help to have a note about this case in the documentation, or prepend it with a symbol that is stripped by Pimcore but it's safe to use in uri/nginx.

dvesh3 committed:
[StaticPageGenerator] Fix generating static content for homepage

dvesh3 commented:
@kingjia90 Good catch! I have adjusted the PR to use `/%home` uri in nginx mapping as this is not possible to create documents with this char prefix and also not possible in pretty url. I have also incorporated the changes suggested by @tmannherz. thanks!

kingjia90 commented:
LGTM and works fine. 
Do we need to adapt the Apache .htaccess documentation as well?

dvesh3 committed:
[StaticPageGenerator] Fix generating static content for homepage

dvesh3 commented:
> LGTM and works fine. Do we need to adapt the Apache .htaccess documentation as well?

Done.
